### PR TITLE
Tj report backend changes for manager report

### PR DESCRIPTION
**Pull request by hemantpurohit1**

### CODE REVIEW

File: tjreports/administrator/views/reports/view.html.php [hunk 1]
```diff
@@ -46,19 +46,38 @@ class TjreportsViewReports extends JViewLegacy
...
 		{
-			JError::raiseError(500, JText::_('JERROR_ALERTNOAUTHOR'));
+			require_once $file;
+
+			$prefix = ucfirst(str_replace('com_', '', $component));
+			$cName = $prefix . 'Helper';
+
+			if (class_exists($cName))
+			{
+				$canDo = TjlmsHelper::getActions();
```
Comment (coolbung) [inline tjreports/administrator/views/reports/view.html.php]: Is this correct ?  Where are you using $cName ?

Also, can you explain a little what is happening here ?

File: tjreports/site/views/reports/view.html.php [hunk 2]
```diff
@@ -39,8 +39,11 @@ class TjreportsViewReports extends JViewLegacy
 	public function display($tpl = null)
 	{
 		$canDo = TjreportsHelpersTjreports::getActions();
-		$user = JFactory::getUser();
-		$user_id = $user->id;
+		$this->user       = JFactory::getUser();
+		$this->user_id    = $this->user->id;
+
+		$user = $this->user;
+		$user_id = $this->user_id;
```
Comment (coolbung) [inline tjreports/site/views/reports/view.html.php]: Why do you need so many additional variables ?

Comment (twsvaishali) [inline tjreports/site/views/reports/view.html.php]: @hemantpurohit1 use $this->user and $this->user_id

remove
$user = $this->user;
$user_id = $this->user_id;

File: tjreports/administrator/tjreports.xml [hunk 3]
```diff
@@ -32,11 +32,12 @@
         <filename>index.html</filename>
         <filename>tjreports.php</filename>
         <filename>controller.php</filename>
-        <filename>router.php</filename>
+        <filename>router1.php</filename>
```
Comment (twsvaishali) [inline tjreports/administrator/tjreports.xml]: Remove this line and also router1.php file

File: tjreports/administrator/views/reports/view.html.php [hunk 4]
```diff
@@ -145,7 +166,9 @@ public function display($tpl = null)
 	protected function addToolbar()
 	{
 		// Old code
-		$extension = JFactory::getApplication()->input->get('extension', '', 'word');
+		$reportsModel = $this->getModel();
+		$extension = $reportsModel->getState('extension');
```
Comment (twsvaishali) [inline tjreports/administrator/views/reports/view.html.php]: @hemantpurohit1  Use the $extension set in the display method.

Comment (twsvaishali) [inline tjreports/administrator/views/reports/view.html.php]: in display, $this->extension and use here

**Pull request merged**